src/HvLightPipe.c

```diff
@@ -20,11 +20,13 @@
 #include <xmmintrin.h>
 #define hv_sfence() _mm_sfence()
 #elif __arm__ || HV_SIMD_NEON
-  #if __ARM_ACLE
+  #ifdef __ARM_ACLE
     #include <arm_acle.h>
     // https://msdn.microsoft.com/en-us/library/hh875058.aspx#BarrierRestrictions
     // http://doxygen.reactos.org/d8/d47/armintr_8h_a02be7ec76ca51842bc90d9b466b54752.html
     #define hv_sfence() __dmb(0xE) /* _ARM_BARRIER_ST */
+  #elif defined ARM_CORTEX
+    #define hv_sfence() __ASM volatile ("dmb 0xF":::"memory")
```

Comment:
Thanks @pingdynasty, super interesting! Is there a reason why line 27 didn't work for you? And is there a reason why you use `dmb 0xF` (full read/write barrier: `_ARM_BARRIER_ST `) instead of `dmb 0xE` (write barrier: `_ARM_BARRIER_ST`)?